app/src/main/java/com/geeksville/mesh/ui/MapFragment.kt

```diff
@@ -1,680 +1,316 @@
+                onNodesChanged(nodes.values)
+                drawOverlays()
+            }
+            zoomToNodes(mapController)
+        }
+    }
 
+    private fun chooseMapStyle() {
+        /// Prepare dialog and its items
+        val builder = MaterialAlertDialogBuilder(context!!)
+        builder.setTitle(getString(R.string.preferences_map_style))
+        val mapStyles by lazy { resources.getStringArray(R.array.map_styles) }
+
+        /// Load preferences and its value
+        val prefs = UIViewModel.getPreferences(context!!)
+        val editor: SharedPreferences.Editor = prefs.edit()
+        val mapStyleInt = prefs.getInt(mapStyleId, 1)
+        debug("mapStyleId from prefs: $mapStyleInt")
+
+        builder.setSingleChoiceItems(mapStyles, mapStyleInt) { dialog, which ->
+            debug("Set mapStyleId pref to $which")
+            editor.putInt(mapStyleId, which)
+            editor.apply()
+            dialog.dismiss()
+            map.setTileSource(loadOnlineTileSourceBase())
+        }
+        val dialog = builder.create()
+        dialog.show()
+    }
 
     private fun onNodesChanged(nodes: Collection<NodeInfo>) {
         val nodesWithPosition = nodes.filter { it.validPosition != null }
 
         /**
-         * Using the latest nodedb, generate geojson features
+         * Using the latest nodedb, generate GeoPoint
          */
-        fun getCurrentNodes(): FeatureCollection {
-            // Find all nodes with valid locations
-
-            val locations = nodesWithPosition.map { node ->
+        // Find all nodes with valid locations
+        fun getCurrentNodes(): List<Marker> {
+            val mrkr = nodesWithPosition.map { node ->
                 val p = node.position!!
                 debug("Showing on map: $node")
-
-                val f = Feature.fromGeometry(
-                    Point.fromLngLat(
-                        p.longitude,
-                        p.latitude
-                    )
-                )
+                val f = GeoPoint(p.latitude, p.longitude)
+                lateinit var marker: MarkerWithLabel
                 node.user?.let {
-                    f.addStringProperty("name", it.longName + " " + formatAgo(p.time))
+                    val label = it.longName + " " + formatAgo(p.time)
+                    marker = MarkerWithLabel(map, label)
+                    marker.title = label
+                    marker.setAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_CENTER)
+                    marker.position = f
+                    marker.icon = ContextCompat.getDrawable(
+                        requireActivity(),
+                        R.drawable.ic_twotone_person_pin_24
+                    )
                 }
-                f
+                marker
             }
+            return mrkr
+        }
+        nodePositions = getCurrentNodes()
+    }
+
+    private fun drawOverlays() {
+        map.overlayManager.overlays().clear()
```

Comment:
Temporary solution. Ideally we should only update the locations of each node, rather than re-draw each layer anytime there is an update

Reply:
Not sure if I will be able to change this. It does seem lightweight enough to re-draw the top layers on update.

Reply:
> It does seem lightweight enough to re-draw the top layers on update.

agreed.